go/cmd/dolt/commands/cherry-pick.go

```diff
@@ -109,173 +113,149 @@ func (cmd CherryPickCmd) Exec(ctx context.Context, commandStr string, args []str
-	} else {
-		commitParams := []string{"-m", commitMsg}
-		res := CommitCmd{}.Exec(ctx, "commit", commitParams, dEnv, cliCtx)
-		if res != 0 {
-			return errhand.BuildDError("dolt commit failed").AddCause(err).Build()
+		// if we have a hash and all 0s, then the cherry-pick succeeded
+		if len(commitHash) > 0 && dataConflicts == 0 && schemaConflicts == 0 && constraintViolations == 0 {
+			succeeded = true
 		}
 	}
 
-	return nil
-}
+	if succeeded {
+		// on success, print the commit info
+		commit, err := getCommitInfo(queryist, sqlCtx, commitHash)
+		if err != nil {
+			return fmt.Errorf("error: failed to get commit metadata for ref '%s': %v", commitHash, err)
+		}
 
-// mergeCherryPickedCommit executes a merge to cherry-pick the specified ref specification from
-// |cherryStr| and apply it to the specified |workingRoot| in this |dEnv|. The MergeResult is
-// returned, along with the commit message for the specified |cherryStr|, and any error encountered.
-func mergeCherryPickedCommit(ctx context.Context, dEnv *env.DoltEnv, workingRoot *doltdb.RootValue, cherryStr string) (*merge.Result, string, error) {
-	tmpDir, err := dEnv.TempTableFilesDir()
-	if err != nil {
-		return nil, "", err
-	}
-	opts := editor.Options{Deaf: dEnv.BulkDbEaFactory(), Tempdir: tmpDir}
+		cli.ExecuteWithStdioRestored(func() {
+			pager := outputpager.Start()
+			defer pager.Stop()
 
-	cherrySpec, err := doltdb.NewCommitSpec(cherryStr)
-	if err != nil {
-		return nil, "", err
-	}
-	headRef, err := dEnv.RepoStateReader().CWBHeadRef()
-	if err != nil {
-		return nil, "", err
-	}
-	cherryCm, err := dEnv.DoltDB.Resolve(ctx, cherrySpec, headRef)
-	if err != nil {
-		return nil, "", err
-	}
+			printCommitInfo(pager, 0, false, "auto", commit)
+		})
 
-	if len(cherryCm.DatasParents()) > 1 {
-		return nil, "", errhand.BuildDError("cherry-picking a merge commit is not supported.").Build()
-	}
-	if len(cherryCm.DatasParents()) == 0 {
-		return nil, "", errhand.BuildDError("cherry-picking a commit without parents is not supported.").Build()
+		return nil
+	} else {
+		// this failure could only have been caused by constraint violations or conflicts during cherry-pick
+		return ErrCherryPickConflictsOrViolations.New()
 	}
+}
 
-	cherryCM, err := cherryCm.GetCommitMeta(ctx)
+func cherryPickAbort(queryist cli.Queryist, sqlCtx *sql.Context) error {
+	q := "call dolt_merge('--abort')"
+	_, err := GetRowsForSql(queryist, sqlCtx, q)
 	if err != nil {
-		return nil, "", err
+		errorText := err.Error()
+		switch errorText {
+		case "fatal: There is no merge to abort":
+			return fmt.Errorf("error: There is no cherry-pick merge to abort")
+		default:
+			return err
+		}
 	}
-	commitMsg := cherryCM.Description
+	return nil
+}
 
-	cherryRoot, err := cherryCm.GetRootValue(ctx)
-	if err != nil {
-		return nil, "", err
-	}
+func getDoltStatus(queryist cli.Queryist, sqlCtx *sql.Context) (hasStagedChanges bool, hasUnstagedChanges bool, err error) {
```

Comment:
This code in particular is probably not very useful on its own, since it only returns two booleans that indicate staged/working changes are present. Other commands that need to consult `dolt_status` are pulling more data from that table.